test/physics/em/BetheHeitlerInteractor.test.cc

```diff
@@ -0,0 +1,228 @@
+        energy1.push_back(result.secondaries[0].energy.value());
+        energy2.push_back(result.secondaries[1].energy.value());
+    }
+
+    EXPECT_EQ(2 * num_samples, this->secondary_allocator().get().size());
+
+    // Note: these are "gold" values based on the host RNG.
+    const double expected_energy1[] = {
+        3.39788055741559, 51.5987452963335, 92.643071139302, 37.1648912184596};
+    const double expected_energy2[] = {
+        96.6021194425844, 48.4012547036665, 7.35692886069806, 62.8351087815404};
+    const double expected_angle[] = {0.925918740331658,
+                                     0.999654206663411,
+                                     0.989152414479486,
+                                     0.999424622760949};
+
+    EXPECT_VEC_SOFT_EQ(expected_energy1, energy1);
+    EXPECT_VEC_SOFT_EQ(expected_energy2, energy2);
+    EXPECT_VEC_SOFT_EQ(expected_angle, angle);
+
+    // Next sample should fail because we're out of secondary buffer space
+    {
+        Interaction result = interact(rng_engine);
+        EXPECT_EQ(0, result.secondaries.size());
+        EXPECT_EQ(celeritas::Action::failed, result.action);
+    }
+}
+
+// TEST_F(BetheHeitlerInteractorTest, stress_test)
+// {
+//     RandomEngine& rng_engine = this->rng();
+
+//     const unsigned int num_samples = 8;
+
+//     // Loop over a set of incident gamma energies
+//     for (double inc_e : {1.5, 10.0, 100.0})
+//     {
+//         this->set_inc_particle(pdg::gamma(), MevEnergy{inc_e});
+
+//         // Loop over several incident directions
+//         for (const Real3& inc_dir :
+//              {Real3{0, 0, 1}, Real3{1, 0, 0}, Real3{1e-9, 0, 1}, Real3{1, 1,
+//              1}})
+//         {
+//             SCOPED_TRACE("Incident direction: " + to_string(inc_dir));
+//             this->set_inc_direction(inc_dir);
+//             this->resize_secondaries(2 * num_samples);
+
+//             // Get the ElementView
+//             const celeritas::ElementView element(
+//                 this->material_track().material_view().element_view(
+//                     celeritas::ElementComponentId{0}));
+
+//             // Create interactor
+//             BetheHeitlerInteractor interact(pointers_,
+//                                             this->particle_track(),
+//                                             this->direction(),
+//                                             this->secondary_allocator(),
+//                                             element);
+
+//             // Loop over many particles
+//             for (unsigned int i = 0; i < num_samples; ++i)
+//             {
+//                 Interaction result = interact(rng_engine);
+//                 SCOPED_TRACE(result);
+//                 this->sanity_check(result);
+//             }
+//             EXPECT_EQ(2 * num_samples,
+//                       this->secondary_allocator().get().size());
+//         }
+//     }
+// }
```

Comment:
I think this is the last thing that needs to be added! Since we're pretty confident for now that the implementation is almost equivalent to Geant4's, let's do like we did with Amanda's PR and disable the conservation check in this unit test. Can you also add to this test the RNG invocation counter like is in the PE/KN tests?

Reply:
Regarding the conservation checks: I thought the energy and momentum conservation checks were split into separate functions in a recent PR. #100? Not a big deal -- I can add the change here.
Also, we do have energy conservation in the Bethe-Heitler, so if the separate conservation checks are done in their own respective functions (and these are `public`), I can keep the energy check.

Reply:
Oh I thought you'd said you'd make that change on this branch. Let's just do the momentum/energy separation on a subsequent branch and tackle both the new physics interactors at once. Maybe that can be something we can work on wednesday?

Reply:
Wednesday it is.